src/tarantool.c

```diff
@@ -251,7 +251,9 @@ static int __tarantool_connect(tarantool_object *t_obj) {
 	TSRMLS_FETCH();
 	tarantool_connection *obj = t_obj->obj;
 	int status = SUCCESS;
-	long count = TARANTOOL_G(retry_count);
+	/* retry_count always 1 more than count of tries, since retry count
+	 * is number of retries*/
+	long count = TARANTOOL_G(retry_count) + 1;
```

Comment:
I have two points here:

1. Backward compatibility. We change the sense of the option.
2. What we want the option to mean? Tries count or tries count after the first one? I think the first (old behaviour). The real issue was the error message. I think we should get the old behaviour back and give a user an error message indicating the zero is not the valid value for that option.

Reply:
@Totktonada I think that `0` is still a valid value for `retry_count` in case you want to *disable* the retrying functionality. But, in overal, I would complitelly drop this ini setting (and others as well) for the reason I stated in https://github.com/tarantool/tarantool-php/issues/75.

Reply:
My point is that I would like to have a way to disable retries. I don't really care if it starts from 1 or 0 as long as it can be disabled entirely.

Reply:
> 0 is not valid 'tries count'. 

Btw, I was talking about the `tarantool.retry_count` setting. And it's called `*re*try_count`, so I don't see your confusion, as I read it, can be one insufficient call and `0` *retry* calls afterward.

Reply:
Two interpretations are possible, yep. I think the first one is more intuitive. If Eugine would agree with the second one, I will not insist on mine one. But while the interpretation is not obvious it should be clearly stated in the documentation (with any implementation we'll choose). I also propose to write explicitly what option will give 'try just one time' behaviour in the doc.

Reply:
I think that confusion comes from `try_count` vs `retry_count`. You were talking about former and I about latter ). I dug a bit more on how others interpret it and found these: 

https://docs.aws.amazon.com/general/latest/gr/api-retries.html
> Use the ClientConfiguration class and provide a maxErrorRetry value of 0 to turn off the retries. 

https://media.datadirect.com/download/docs/odbc/allodbc/index.html#page/odbc/connection-retry-count.html
> Connection Retry Count
> If set to 0, the driver does not try to connect after the initial unsuccessful attempt.

http://php.net/manual/en/function.sybase-deadlock-retry-count.php

> retry_count
> -1 | Retry forever (default)
>  0 | Do not retry
>  n | Retry n times

https://github.com/NoRedInk/rspec-retry/issues/11

Reply:
Thank you!

So I okay with the new implementation, but the doc need to be clarified within this PR.

Reply:
README updated.

Reply:
To be honest, now I'm not sure again. A behaviour change should be cleanly motivated. We should decide whether the change is bugfix (make the implementation work as it is described in the README) or behaviour change to better fit outside world.

README states:

> * `tarantool.retry_count` - Count of retries for connecting (defaults: 1, can be changed in runtime)

And everything again blocks on how we'll interpret the word `retry`. And if we'll see from the documentation point of view (a user usually lean on the documentation, it is our contract) the default value should not be changed. That means that actual default behaviour will be changed. Oh...

About outside world: I see that when an option is named `retry_count` or so it usually means amount of attempts after the first one. However there are exceptions: `--retries` in [parallel](https://www.gnu.org/software/parallel/parallel_tutorial.html) and our [tarantool-java](https://github.com/tarantool/tarantool-java/blob/9936054e82f0ef109e5637ec64f6d0424c61e799/README.md#client-config-options) connector (default: 3). However, say, [tarantool-php/client](https://github.com/tarantool-php/client/blob/0241eb3ea7f17f7fd2d512ac08d2383e72db233e/README.md#array-of-options) has `max_retries` that means amount of attempts after the initial unsuccessful one (default: 0).

Ugh… I'll come back tomorrow.

Reply:
I pushed the alternative in PR #166. No decision what to choose yet.

Reply:
> However there are exceptions: --retries in parallel 

From `man parallel`:

```
--retries n
    If a job fails, retry it on another computer on which it has not failed. Do this n times. ...
```

Reply:
> > However there are exceptions: --retries in parallel
> 
> From `man parallel`:
> 
> ```
> --retries n
>     If a job fails, retry it on another computer on which it has not failed. Do this n times. ...
> ```

The link above shows example that looks opposite.

Reply:
> The link above shows example that looks opposite.

I know, I just wanted to show that maybe `parallel` is not the best example to follow because its behavior does not reflect what is stated in its manual.

Reply:
Wow, Hi guys. IMHO @rybakit is damn right and the name of the option not match the content. But, backward compatibility is a very strong argument. I think updating the description is the optimal decision. I vote for #166 .